api/tagset.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2015 Square Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"bytes"
+	"regexp"
+	"sort"
+)
+
+// TagSet is the set of key-value pairs associated with a given metric.
+type TagSet map[string]string
+
+// NewTagSet creates a new instance of TagSet.
+func NewTagSet() TagSet {
+	return make(map[string]string)
+}
+
+// Equals check whether two tags are equal.
+func (left TagSet) Equals(right TagSet) bool {
+	if len(left) != len(right) {
+		return false
+	}
+	for k := range left {
+		_, ok := right[k]
+		if !ok || left[k] != right[k] {
+			return false
+		}
+	}
+	return true
+}
+
+// Merge two tagsets, and return a new tagset.
+// If keys conflict, the first tag set is preferred.
+func (tagSet TagSet) Merge(other TagSet) TagSet {
+	result := NewTagSet()
+	for key, value := range other {
+		result[key] = value
+	}
+	for key, value := range tagSet {
+		result[key] = value
+	}
+	return result
+}
+
+//TODO(cchandler): What is this? Should it be moved or deleted?
```

Comment:
Yeah, the note was made a few days ago before I got that far. I'll remove my note :-)